src/components/ImportJobs/ImportJobs.js

```diff
@@ -356,7 +356,8 @@ export class ImportJobs extends Component {
             cancelLabel={<FormattedMessage id="ui-data-import.cancel" />}
             onConfirm={() => {
               this.hideFilesExtensionsModal();
-              openDialogWindow();
+              // Ensure that any changes or state updates are completed before the dialog is opened
+              setTimeout(() => openDialogWindow(), 0);
```

Comment:
Could you simplify it to
```suggestion
              setTimeout(openDialogWindow, 0);
```